src/libraries/System.Private.CoreLib/src/System/IO/DriveInfoInternal.Windows.cs

```diff
@@ -24,7 +24,7 @@ public static string[] GetLogicalDrives()
             int count = BitOperations.PopCount((uint)drives);
 
             string[] result = new string[count];
-            Span<char> root = stackalloc char[] { 'A', ':', '\\' };
+            Span<char> root = ['A', ':', '\\'];
```

Comment:
According to [sharplab, main 12 Oct 2023](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTCDMACZPAMQHsSAhCAJz0wG9M8mD8BjAC2rwHkqoBzWBAA2ACgCUjZgwzM5eAMoAHCDAA8HagD48VMgBc8AXjwBnfRFYBrEcJKs8mqgG0Aunjp4A5AEEvAGm8QAO8AHVCvPABfAG4peQIAdl0DZwA9FFc42WYo+NocplxHThoABQAlCXyZBKZlVQ1SnT0SQxNnXxCvYMCvcK8s/LkEZNb9dMzsuTzCgpG2UrxKKhxqudq6ioBTCAATbhhhAE8G9ScWg2M8Tr8+3rCIobmRsdSM55nMWaA===) this allocates.

Looks like the collection expression only avoid the allocation for ROS, but a ROS can't be used here.